kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/AbstractProducerMetrics.java

```diff
@@ -0,0 +1,262 @@
+
+  static final String REQUEST_COUNT_WINDOWED_METRIC_NAME = "request-count-windowed";
+  private static final String REQUEST_COUNT_WINDOWED_METRIC_DOC =
+      "The total number of requests sent within the given window.";
+
+  // response
+  static final String RESPONSE_RATE_METRIC_NAME = "response-rate";
+  private static final String RESPONSE_RATE_METRIC_DOC =
+      "The average number of responses sent per second.";
+
+  static final String RESPONSE_COUNT_WINDOWED_METRIC_NAME = "response-count-windowed";
+  private static final String RESPONSE_COUNT_WINDOWED_METRIC_DOC =
+      "The total number of responses sent in the given window.";
+
+  // errors
+  static final String RECORD_ERROR_COUNT_WINDOWED_METRIC_NAME = "error-count-windowed";
+  private static final String RECORD_ERROR_COUNT_WINDOWED_METRIC_DOC =
+      "The total number of record sends that resulted in errors in the given window.";
+
+  protected static final String RECORD_ERROR_RATE_METRIC_NAME = "record-error-rate";
+  private static final String RECORD_ERROR_RATE_METRIC_DOC =
+      "The average per-second number of record sends that resulted in errors.";
+
+  // latency
+  static final String REQUEST_LATENCY_AVG_METRIC_NAME = "request-latency-avg";
+  private static final String REQUEST_LATENCY_AVG_METRIC_DOC = "The average request latency";
+
+  static final String REQUEST_LATENCY_MAX_METRIC_NAME = "request-latency-max";
+  private static final String REQUEST_LATENCY_MAX_METRIC_DOC = "The max request latency";
+
+  static final String REQUEST_LATENCY_PCT_METRIC_PREFIX = "request-latency-";
+  private static final String REQUEST_LATENCY_PCT_METRIC_DOC = "Request latency percentiles.";
+
+  protected Metrics metrics;
+  protected String jmxPrefix;
+  protected String requestSensorName;
+  protected String requestSizeSensorName;
+  protected String responseSensorName;
+  protected String recordErrorSensorName;
+  protected String requestLatencySensorName;
+  protected static final Map<String, String> metricsTags = new HashMap<>();
+
+  protected void addMetricsTag(String tagName, String tagValue) {
+    metricsTags.put(tagName, tagValue);
+  }
+
+  protected static Metrics createMetrics(RestMetricsContext metricsContext, Time time) {
+    JmxReporter reporter = new JmxReporter();
+    reporter.contextChange(metricsContext);
+    return new Metrics(
+        new MetricConfig()
+            .samples(NUM_SAMPLES)
+            .timeWindow(SAMPLE_WINDOW_MS, TimeUnit.MILLISECONDS)
+            .recordLevel(RECORDING_LEVEL)
+            .tags(metricsTags),
+        singletonList(reporter),
+        time,
+        metricsContext);
+  }
+
+  protected void setupSensors() {
+    // request metrics
+    setupRequestSensor();
+    setupRequestSizeSensor();
+
+    // response metrics
+    setupResponseSensor();
+    setupRecordErrorSensor();
+    setupRequestLatencySensor();
+
+    log.info("Successfully registered kafka-rest produce metrics with JMX");
```

Comment:
nit: Maybe move this out of this common registration point back to `ProducerMetricsImpl` ctor and add a similar log for the private implementation in the other repo.